api2/pkg/auth/middleware.go

```diff
@@ -23,47 +24,59 @@ func (h *Handler) Authenticate() func(handler http.Handler) http.Handler {
 
 			// Retrieve from store if available
 			accessToken := h.Store.GetString(ctx, AccessTokenKey)
+			isInStore := len(accessToken) != 0
 
 			// If not, fallback to Authorization header
 			if len(accessToken) == 0 {
-
 				bearerStr := req.Header.Get("Authorization")
 				if strings.HasPrefix(bearerStr, "Bearer ") {
 					accessToken = strings.TrimPrefix(bearerStr, "Bearer ")
-				} else {
-					doNext(req, false)
-					return
 				}
+			}
 
-				jkws, err := h.KeycloakClient.GetPublicKeys()
-				if err != nil {
-					doNext(req, false)
-					return
-				}
+			if len(accessToken) == 0 {
```

Comment:
Since we have `isInStore`, could this be `!isInStore` instead?

Reply:
Not really, because the access token might come from the store or from the Authorization header...